inc/Engine/Preload/Cron/Subscriber.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace WP_Rocket\Engine\Preload\Cron;
+
+use WP_Rocket\Engine\Preload\Admin\Settings;
+use WP_Rocket\Engine\Preload\Controller\PreloadUrl;
+use WP_Rocket\Engine\Preload\Database\Queries\Cache;
+use WP_Rocket\Engine\Preload\Database\Queries\RocketCache;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+class Subscriber implements Subscriber_Interface {
+
+	/**
+	 * Preload settings.
+	 *
+	 * @var Settings
+	 */
+	protected $settings;
+
+	/**
+	 * Db query.
+	 *
+	 * @var Cache
+	 */
+	protected $query;
+
+	/**
+	 * Preload url controller.
+	 *
+	 * @var PreloadUrl
+	 */
+	protected $preload_controller;
+
+	/**
+	 * Creates an instance of the class.
+	 *
+	 * @param Settings   $settings Preload settings.
+	 * @param Cache      $query Db query.
+	 * @param PreloadUrl $preload_controller Preload url controller.
+	 */
+	public function __construct( Settings $settings, Cache $query, PreloadUrl $preload_controller ) {
+		$this->settings           = $settings;
+		$this->query              = $query;
+		$this->preload_controller = $preload_controller;
+	}
+
+	/**
+	 * Return an array of events that this subscriber listens to.
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'rocket_preload_clean_rows_time_event' => 'remove_old_rows',
+			'rocket_load_preload_url'              => 'load_preload_url',
+			'cron_schedules'                       => 'add_interval',
+			'init'                                 => [
+				[ 'schedule_clean_not_commonly_used_rows' ],
+				[ 'schedule_pending_jobs' ],
+			],
+		];
+	}
+
+	/**
+	 * Schedule clean from removing of old urls.
+	 *
+	 * @return void
+	 */
+	public function schedule_clean_not_commonly_used_rows() {
+
+		if (
+			! $this->settings->is_enabled()
+			&&
+			wp_next_scheduled( 'rocket_preload_clean_rows_time_event' )
+		) {
+			wp_clear_scheduled_hook( 'rocket_preload_clean_rows_time_event' );
+
+			return;
+		}
```

Comment:
We can remove this, so that the clean still occurs when the option is disabled to avoid stale data